File: lemur-app/src/main/java/io/github/malczuuu/lemur/app/infra/jackson/JacksonConfiguration.java
package io.github.malczuuu.lemur.app.infra.jackson;

import io.github.malczuuu.lemur.app.domain.player.PlayerStatus;
import org.springframework.boot.jackson.autoconfigure.JsonMapperBuilderCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class JacksonConfiguration {

  @Bean
  public JsonMapperBuilderCustomizer enumLabelMixInCustomizer() {
    return builder -> builder.addMixIn(PlayerStatus.class, EnumLabelMixIn.class);
  }
}

File: lemur-app/src/main/kotlin/io/github/malczuuu/lemur/app/infra/jackson/JacksonConfiguration.kt
package io.github.malczuuu.lemur.app.infra.jackson

import io.github.malczuuu.lemur.app.domain.player.PlayerStatus
import org.springframework.boot.jackson.autoconfigure.JsonMapperBuilderCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class JacksonConfiguration {

    @Bean
    fun enumLabelMixInCustomizer(): JsonMapperBuilderCustomizer = {
        it.addMixIn(PlayerStatus::class.java, EnumLabelMixIn::class.java)
    }
}
